src/core_ocean/shared/mpas_ocn_vmix.F

```diff
@@ -796,7 +796,146 @@ subroutine ocn_vel_vmix_tend_implicit_spatially_variable_mannings(meshPool, bott
+         layerThickness !< Input: thickness at cell center
+
+       real (kind=RKIND), dimension(:), intent(in) :: &
+         bottomDrag !< Input: bottomDrag at cell centeres
+
+        real (kind=RKIND), dimension(:), pointer :: ssh
+
+       real (kind=RKIND), dimension(:), pointer :: bottomDepth
+
+      !-----------------------------------------------------------------
+      !
+      ! input/output variables
+      !
+      !-----------------------------------------------------------------
 
+      real (kind=RKIND), dimension(:,:), intent(inout) :: &
+         normalVelocity             !< Input: velocity
+
+      real (kind=RKIND), dimension(:,:), intent(inout) :: &
+         layerThicknessEdge        !< Input: thickness at edge
+
+      !-----------------------------------------------------------------
+      !
+      ! output variables
+        
+        integer, intent(out) :: err !< Output: error flag
+
+      !-----------------------------------------------------------------
+      !
+      ! local variables
+      !
+      !-----------------------------------------------------------------
+
+      integer :: iEdge, k, cell1, cell2, N, nEdges
+      integer, pointer :: nVertLevels
+      real (kind=RKIND) :: implicitCd
+      integer, dimension(:), pointer :: nEdgesArray
+
+      integer, dimension(:), pointer :: maxLevelEdgeTop
+
+      integer, dimension(:,:), pointer :: cellsOnEdge
+
+      real (kind=RKIND), dimension(:), allocatable :: A, B, C, velTemp
+        
+        err = 0
+
+      if(.not.velVmixOn) return
+
+      call mpas_pool_get_dimension(meshPool, 'nEdgesArray', nEdgesArray)
+      call mpas_pool_get_dimension(meshPool, 'nVertLevels', nVertLevels)
+
+      call mpas_pool_get_array(meshPool, 'maxLevelEdgeTop', maxLevelEdgeTop)
+      call mpas_pool_get_array(meshPool, 'cellsOnEdge', cellsOnEdge)
+
+      nEdges = nEdgesArray( 1 )
+
+      allocate(A(nVertLevels),B(nVertLevels),C(nVertLevels),velTemp(nVertLevels))
+      A(1)=0.0_RKIND
+
+      !$omp do schedule(runtime)
+      do iEdge = 1, nEdges
+        N = maxLevelEdgeTop(iEdge)
+        if (N .gt. 0) then
+        ! Compute A(k), B(k), C(k)
+         ! layerThicknessEdge is computed in compute_solve_diag, and is not
+         ! available yet,
+         ! so recompute layerThicknessEdge here.
+         cell1 = cellsOnEdge(1,iEdge)
+         cell2 = cellsOnEdge(2,iEdge)
+         do k = 1, N
+         layerThicknessEdge(k,iEdge) = 0.5_RKIND * (layerThickness(k,cell1) &
+         +layerThickness(k,cell2))
+         end do
+
+         ! average cell-based implicit bottom drag following the formulation
+         ! in HYCOM (reference : NCOM's formula for 2D case 
+         !cb(i,j) = max(cbmin, (vonk/log(0.5*depth(i,j)/z0))**2 ))
+        implicitCd =max(0.0025,min(0.16/log(0.5*0.5*((bottomDepth(cell1)+bottomDepth(cell2))/0.001)**2.0),0.1))        
+        !! if((implicitCd.ge.0.1).or.(implicitCd.le.0.0001))then
+        !!implicitCd=0.0025
+        !!endif
```

Comment:
May not need the commented lines above....

Reply:
@nairita87 In this line the square should be outside of the log, right? I think it should be
```
implicitCd = max(0.0025, min(0.16/log(0.5*0.5*(bottomDepth(cell1)+bottomDepth(cell2))/0.001)**2.0, 0.1 ) )
```

Reply:
Also do we want to save the drag coefficient, perhaps by updating `bottomDrag` so that it can be used elsewhere, e.g., in GOTM?

Reply:
@qingli411 yes I think you are right. I checked the paper by Oley. The square is over log(zb/z0).

I am confused because, in HYCOM, the code is as follows:
CBP(I,J) = MAX( CBMIN,
     &                      VONK**2/LOG(0.5*DEPTH(I,J)/Z0)**2 )

Yes I will save the drag coefficient